pax-web-runtime/src/main/java/org/ops4j/pax/web/service/internal/ServletEventDispatcher.java

```diff
@@ -135,12 +134,6 @@ public void servletEvent(final ServletEvent event) {
 		}
 		synchronized (listeners) {
 			callListeners(event);
-			Map<String, ServletEvent> events = states.get(event.getBundle());
-			if (events == null) {
-				events = new LinkedHashMap<String, ServletEvent>();
-				states.put(event.getBundle(), events);
-			}
-			events.put(event.getAlias(), event);
```

Comment:
This is actually there for new listeners to retrieve already sent events. What did make you think it needs to be removed. I can see that it might not be a good idea to actually keep a list of bundles in the list. This might be a case where it is better to have this optimized.